kv/kvnemesis: TestKVNemesisSingleNode failed

kv/kvnemesis.TestKVNemesisSingleNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3291445&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3291445&tab=artifacts#/) on master @ [e723a85739a2e1b514f5de3221a2177a3a37d0e6](https://github.com/cockroachdb/cockroach/commits/e723a85739a2e1b514f5de3221a2177a3a37d0e6):


```
Slow failing tests:
TestKVNemesisSingleNode - 0.00s

Slow passing tests:
<none>
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestKVNemesisSingleNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
danhhz marked as alumn{us,a}; ignoring
```
</p>
</details>

/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Hung test after ~2930 runs:

```
=== RUN   TestKVNemesisSingleNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisSingleNode950353309
    test_log_scope.go:74: use -show-logs to present logs inline
E210812 06:10:36.742939 1451 kv/kvserver/queue.go:1213  [n1,replicate] 1  43 replicas failing with "0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster"
E210812 06:20:36.745641 1451 kv/kvserver/queue.go:1213  [n1,replicate] 2  43 replicas failing with "0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster"
E210812 06:30:36.738741 1451 kv/kvserver/queue.go:1213  [n1,replicate] 3  43 replicas failing with "0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster"
panic: test timed out after 40m0s
```

@AlexTalks You've worked a bit with kvnemesis lately, want to have a look? Seems to me like we may be requesting more replicas than there are nodes. Unclear why it's only failing very rarely, but I suppose setting a replication factor of 1 in this test would possibly do the trick.

--- a/pkg/kv/kvnemesis/kvnemesis.go
+++ b/pkg/kv/kvnemesis/kvnemesis.go
@@ -60,8 +60,11 @@ func RunNemesis(
 
 			recCtx, collect, cancel := tracing.ContextWithRecordingSpan(
 				ctx, tracing.NewTracer(), "txn step")
+			buf.Reset()
+			fmt.Fprintf(&buf, "step:")
+			step.format(&buf, formatCtx{indent: `  ` + workerName + ` PRE  `})
+			log.VEventf(recCtx, 2, "%v", buf.String())
 			err := a.Apply(recCtx, &step)
-			log.VEventf(recCtx, 2, "step: %v", step)
 			step.Trace = collect().String()
 			cancel()
 			if err != nil {

